unikernel.ml

```diff
@@ -146,9 +146,28 @@ struct
     Middleware.apply_middleware middlewares
       (fun _reqd ->
         match Middleware.user_of_cookie users now reqd with
-        | Ok user -> f user
-        | Error (`Msg msg) ->
-            Logs.err (fun m -> m "couldn't find user of cookie: %s" msg);
+        | Ok user_cookie -> (
+            let user, cookie = user_cookie in
+            let cookie =
+              { cookie with user_agent = Middleware.user_agent reqd }
+            in
+            let cookies =
+              List.map
+                (fun (cookie' : User_model.cookie) ->
+                  if String.equal cookie.value cookie'.value then cookie
+                  else cookie')
+                user.cookies
+            in
+            let updated_user = User_model.update_user user ~cookies () in
+            Store.update_user !store updated_user >>= function
+            | Ok store' ->
+                store := store';
+                f user_cookie
```

Comment:
this change I don't quite understand... so far (I haven't finished reviewing all changes), what the call site does is `let user, _ = user_cookie in`... Could we get the user only back -- and if we need the cookies as well somewhere, retrieve them by lookup?